### PR TITLE
Mixins

### PR DESCRIPTION
**Pull request by lauriejones**

### CODE REVIEW

File: compiled.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  @include button-size("large");
+}
+
+.v2-button--icon-on-left > .v2-icon {
+  margin-right: 10px;
+}
+
+.v2-button--icon-on-right > .v2-icon {
+  margin-left: 10px;
+}
```
Comment (jameslnewell) [inline compiled.scss]: Love all these mixins!

File: dist/compiled.css [hunk 2]
```diff
@@ -0,0 +1,155 @@
+@font-face {
```
Comment (jameslnewell) [inline dist/compiled.css]: We probably don't want to compile v2-icons in for component, cause v2-icons is listed as a dep, component will include v2-icons resulting in duplicate code.

File: index.css [hunk 3]
```diff
@@ -1,239 +0,0 @@
-/* === Common === */
-.v2-button {
-  cursor: pointer;
-  font-size: 14px;
-  font-weight: 700;
-  font-family: "Open Sans", "Helvetica Neue", Helvetica, sans-serif;
```
Comment (jameslnewell) [inline index.css]: What do you reckon about variablising this in v2-typo?

File: index.css [hunk 4]
```diff
@@ -1,239 +0,0 @@
...
-  border: none;
-  background-color: transparent;
-  display: inline-block;
-  /* === Modifier: Disabled === */
-  /* === Modifier: Size === */
-  /* === Modifier: Icon === */ }
-  .v2-button--is-disabled, .v2-button[disabled] {
-    cursor: default; }
-  .v2-button--medium {
-    font-family: "roboto", "Helvetica Neue", Helvetica, sans-serif;
```
Comment (jameslnewell) [inline index.css]: And this

File: lib/colors.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+$tertiaryBgColorGreen:      transparent;
+$tertiaryBgColorGrey:       transparent;
+$tertiaryBgColorWhite:      transparent;
+
+$textColorGreen:            $color-queen-elizabeth;
+$textColorGrey:             $color-doc;
+$textColorWhite:            $color-white;
+
+
+@function hover-color($color) {
```
Comment (jameslnewell) [inline lib/colors.scss]: i know you didn't originally write these, I wonder if we should prefix these e.g. button-hover-color, button-text-color

File: lib/mixins.scss [hunk 6]
```diff
@@ -0,0 +1,262 @@
+@import "@nib-styles/v2-icons/dist/mixin.scss";
+
+@mixin button-common() {
+  box-sizing: border-box;
+  cursor: pointer;
+  font-size: 14px;
+  font-weight: 700;
+  font-family: "Open Sans", "Helvetica Neue", "Helvetica", sans-serif;
```
Comment (jameslnewell) [inline lib/mixins.scss]: What do you reckon about variablising these font stacks in v2-typo?

File: lib/mixins.scss [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+    &[disabled] {
+      color: rgba($textColorWhite, 0.3);
+    }
+  }
+}
+
+@mixin button-size($size) {
+  @if ($size == "medium") {
+
+    font-family: "roboto", "Helvetica Neue", "Helvetica", sans-serif;
```
Comment (jameslnewell) [inline lib/mixins.scss]: ditto

**Comment by lauriejones** [comment]
ping @jameslnewell

File: README.md [hunk 8]
```diff
@@ -1,23 +1,67 @@
...
+
+## Mixins
 
-See the pattern library or [example.html](example.html) for more examples.
+```scss
+@include button($type, $icon: false, $size: null);
+```
 
-**Note:** Uses `Opens Sans` and `Roboto` fonts from Google:
+- `$type` Required. May be `primary`, `secondary-green`, `secondary-dark-grey`, `secondary-light-grey`, `tertiary-green`, `tertiary-grey` or `tertiary-white`.
```
Comment (jameslnewell) [inline README.md]: @lauriejones Does this need updating?

File: lib/mixins.scss [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+
+}
+
+@mixin button-type($type, $color: null) {
+
+  @if ($type == "primary") {
+
+    background-color: $primaryBgColorOrange;
+
+    &:not(.is-disabled):not([disabled]):hover {
```
Comment (jameslnewell) [inline lib/mixins.scss]: why do the primary ones need this but none of the others?

File: lib/mixins.scss [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+    @include v2-icon($name: $icon, $size: null, $color: null, $rotate: 0, $offset-descenders: true, $after: true);
+
+    &:after {
+      display: inline-block;
+      margin-left: 10px;
+
+      font-style: normal;
+      font-variant: normal;
+      font-weight: normal;
+      font-family: nibdings;
```
Comment (jameslnewell) [inline lib/mixins.scss]: Is there a variable for this in the font repo?

**Comment by jameslnewell** [comment]
Looks good other than the above things

File: lib/mixins.scss (hunk 10 above)
Comment (lauriejones) [inline lib/mixins.scss]: For the nibdings font properties? I don't think so..

File: lib/mixins.scss (hunk 9 above)
Comment (lauriejones) [inline lib/mixins.scss]: True. Mustn't be needed then, will remove. :)

**Comment by jameslnewell** [comment]
With the icon size being the same size as the text, is the offset-descenders too big? It looks a bit like its aligned with the bottom of the button.

Other than that, good to go! Merge, bump the major and publish!!! :tada:

**Pull request merged**